### DISCUSSION TITLE
Card layout not working on Pivot's Relation Manager

### DISCUSSION THREAD

**Discussion by billydekid** [filamentphp/filament#6495] [Help] [answered]
Hi,

I have a BelongsToMany relationship with pivot table on Relation Manager.
The problem was I'm unable to apply Card layout of form component of a Relation Manager.
The Card layout does not work on modal page when I attach a record.

Has anyone having same problem?

EducationsRelationManager.php

    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Card::make()->schema([
                    Forms\Components\Select::make('education_id')
                        ->preload()
                        ->options(Education::query()->pluck('education_level', 'id'))
                        ->required(),
                    Forms\Components\TextInput::make('education_name')
                        ->label('School/Institute name')
                        ->required()
                        ->maxLength(255),
                    Forms\Components\TextInput::make('major')
                        ->required()
                        ->maxLength(255),
                    Forms\Components\TextInput::make('score')
                        ->required()
                        ->maxLength(255),
                    Forms\Components\TextInput::make('start_year')
                        ->numeric()
                        ->maxLength(4)
                        ->required()
                        ->maxLength(255),
                    Forms\Components\TextInput::make('end_year')
                        ->numeric()
                        ->maxLength(4)
                        ->required()
                        ->maxLength(255),
                ])
            ]);
    }

[bayu]

**Comment by billydekid** [comment]
Apologize, I put wrong code parts. It should be on AttachAction of headerActions() properties.
But I still don't know how to put the Card layout on it.

            ->headerActions([
                Tables\Actions\CreateAction::make(),

                Tables\Actions\AttachAction::make()
                    ->preloadRecordSelect()
                    ->label('Add education')
                    ->form(fn (AttachAction $action): array => [
                        $action->getRecordSelect(),
                        Forms\Components\TextInput::make('education_name')
                            ->label('School/Institute name')
                            ->required(),
                        Forms\Components\TextInput::make('major')->required(),
                        Forms\Components\TextInput::make('score')->required(),
                        Forms\Components\TextInput::make('start_year')->numeric()->maxLength(4)->required(),
                        Forms\Components\TextInput::make('end_year')->numeric()->maxLength(4)->required(),
                    ])
                    ->modalHeading('Add education'),
            ])

**Reply by danharrin** [reply, marked as answer]
```php
        ->headerActions([
            Tables\Actions\CreateAction::make(),

            Tables\Actions\AttachAction::make()
                ->preloadRecordSelect()
                ->label('Add education')
                ->form(fn (AttachAction $action): array => [
                    Card::make([
                        $action->getRecordSelect(),
                        Forms\Components\TextInput::make('education_name')
                            ->label('School/Institute name')
                            ->required(),
                        Forms\Components\TextInput::make('major')->required(),
                        Forms\Components\TextInput::make('score')->required(),
                        Forms\Components\TextInput::make('start_year')->numeric()->maxLength(4)->required(),
                        Forms\Components\TextInput::make('end_year')->numeric()->maxLength(4)->required(),
                    ]),
                ])
                ->modalHeading('Add education'),
        ])
```

**Reply by billydekid** [reply]
Many thanks @danharrin
